packages/api/src/app-router.ts

```diff
@@ -143,6 +155,24 @@ export default async function makeApp(params: CliArgs) {
     req.stripe = stripe;
     next();
   });
+  app.use(
+    authenticateWithCors({
+      cors: {
+        anyOriginPathPrefixes: [pathJoin("/", httpPrefix, "/asset/upload/")],
```

Comment:
Hey @anarkrypto! This line here is the one that will make these code changes have exactly the effect that you suggested!

The whole point of the separate `/request-upload` from the `/upload/$token` APIs was that users of the API would not need to proxy the file contents themselves. The `/upload` API doesn't even require authentication, so this behavior of restricting CORS was certainly a bug. 

I've also been testing this PR for the past week and I'm pretty confident that it is good to go. So I'll be merging it tomorrow (just after adding some docs) and will be hopefully deploying it on tuesday! Rest assured that you won't have to deal with file upload traffic from your backend integrating with our VOD API :)

And off course, thanks a lot for your input and detailed issue! If I hadn't already have this PR close to merging I'd certainly work on a hotfix just for your use case, knowing you were blocked/affected by this behavior. Cheers!